src/Surfnet/StepupMiddleware/ApiBundle/Controller/AuditLogController.php

```diff
@@ -31,11 +32,16 @@ public function secondFactorAuditLogAction(Request $request, Institution $instit
     {
         $this->denyAccessUnlessGranted(['ROLE_RA']);
 
+        $identityId = $request->get('identityId');
+        if (empty($identityId)) {
+            throw new BadApiRequestException(['This API-call MUST include the identityId as get parameter']);
```

Comment:
> **TL;DR** IMO, [update](https://github.com/SURFnet/Stepup-Middleware/blob/16550f3852f98a9e7d36c5b3cfdc20d7565abc72/src/Surfnet/StepupMiddleware/ApiBundle/Resources/config/routing.yml#L79-L80) route requirement to be a UUID, remove check in controller, and URL parameter would be nice, but register as tech debt.

I'm not sure whether Symfony enforces a requirement when the requirement is not a URL parameter, but rather a query parameter. We should verify that and [update](https://github.com/SURFnet/Stepup-Middleware/blob/16550f3852f98a9e7d36c5b3cfdc20d7565abc72/src/Surfnet/StepupMiddleware/ApiBundle/Resources/config/routing.yml#L79-L80) the requirement to a UUID. Since the audit log is per-identity, making the identity ID a URL parameter seems the logical choice, but then again, I remember us making conscious decision not to. However, I don't think we should go and change the route URL now, and tech debt would be fine.

Since we rely on the routing definition to impose certain requirements before routing the request to the controller action (for example, `Accept:application/json`), a query parameter requirement in the routing definition would suffice and an extra check would not be needed in the controller action.